### ISSUE TITLE
Instance autoScaling with instanceSizeName

### ISSUE THREAD

**Issue by tetianakravchenko** [mongodb/mongodb-atlas-kubernetes#317]
Similar to https://github.com/mongodb/mongodb-atlas-kubernetes/issues/316:
There is the similar issue with `instanceSize`: if instance autoScaling is enabled and `providerSettings.instanceSizeName` is defined, operator bring cluster to the size defined in `providerSettings.instanceSizeName`.

example:
```
autoScaling:
  compute:
    enabled: true
    scaleDownEnabled: true
providerSettings:
  autoScaling:
    compute:
      maxInstanceSize: M30
      minInstanceSize: M10
  instanceSizeName: M10
  providerName: GCP
  regionName: WESTERN_EUROPE
```
Then I've resized instance manually to M20, with the next changes of AtlasCluster CR (added `providerBackupEnabled: false`, but the actual change is not important here):
```
DEBUG	controllers.AtlasCluster	Clusters are different:   mongodbatlas.Cluster{
  	... // 16 identical fields
  	Paused:                &false,
  	PitEnabled:            &false,
- 	ProviderBackupEnabled: &true,
+ 	ProviderBackupEnabled: &false,
  	ProviderSettings: &mongodbatlas.ProviderSettings{
  		... // 2 identical fields
  		DiskTypeName:     "",
  		EncryptEBSVolume: nil,
- 		InstanceSizeName: "M20",
+ 		InstanceSizeName: "M10",
  		ProviderName:     "GCP",
  		RegionName:       "WESTERN_EUROPE",
  		... // 2 identical fields
  	},
```
operator resize it back to `M10`.

Also, according to the https://docs.atlas.mongodb.com/reference/api/clusters-create-one/ `providerSettings.instanceSizeName` is a required field, so I am getting:
```
Error: UPGRADE FAILED: error validating "": error validating data: ValidationError(AtlasCluster.spec.providerSettings): missing required field "instanceSizeName" in com.mongodb.atlas.v1.AtlasCluster.spec.providerSettings
```
when I remove `instanceSizeName` from config.

What is the expected way of dealing with instance autoScaling?

**Comment by tetianakravchenko** [comment]
@jasonmimick any updates on this issue?

**Comment by jasonmimick** [comment]
Fixed by #322
